src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java

```diff
@@ -129,6 +129,42 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
         verify(caseEventMessageRepository, never()).getNumberOfMessagesReceivedInLastHour(any());
     }
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:scripts/insert_case_event_messages_for_recived_messages_test.sql"})
+    void test_health_reports_up_if_received_messages_in_last_hour() throws Exception {
+
+        // GIVEN
+        setClock(LocalDateTime.of(2024, 4, 2, 14,15));
+
+        // THEN
+        mockMvc.perform(get("/health"))
+            .andExpect(
+                jsonPath("$.components.ccdMessagesReceived.status")
+                    .value(UP.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(MESSAGES_RECEIVED, "test")));
+
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:scripts/insert_case_event_messages_for_recived_messages_test.sql"})
+    void test_health_reports_down_if_not_received_messages_in_last_hour() throws Exception {
+
+        // GIVEN
+        setClock(LocalDateTime.of(2024, 4, 2, 15,15));
+
+        // THEN
+        mockMvc.perform(get("/health"))
+            .andExpect(
+                jsonPath("$.components.ccdMessagesReceived.status")
+                    .value(DOWN.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(NO_MESSAGES_RECEIVED, "test")));
+
+    }
```

Comment:
Did we also want to check the timezone issue in integration tests too or is this covered elsewhere?

FYI @BenNzewi

Reply:
I added 2 integration tests for timesone scenarios Ravi. Thank you.